src/backend/ManagementHub.Storage/Contexts/Tournament/DbTournamentContextProvider.cs

```diff
@@ -850,4 +866,131 @@ public async Task RemoveTeamParticipantAsync(
 		this.logger.LogInformation("Removed team {TeamId} as participant from tournament {TournamentId}",
 			teamId, tournamentId);
 	}
+
+	public async Task UpdateParticipantRosterAsync(
+		TournamentIdentifier tournamentId,
+		TeamIdentifier teamId,
+		RosterUpdateData rosterData,
+		CancellationToken cancellationToken = default)
+	{
+		var tournamentIdString = tournamentId.ToString();
+
+		// Get participant with roster entries
+		var participant = await this.dbContext.TournamentTeamParticipants
+			.Include(p => p.RosterEntries)
+			.Where(p => p.Tournament.UniqueId == tournamentIdString && p.TeamId == teamId.Id)
+			.FirstOrDefaultAsync(cancellationToken);
+
+		if (participant == null)
+		{
+			throw new NotFoundException($"Team {teamId} is not a participant of tournament {tournamentId}");
+		}
+
+		// Collect all user IDs
+		var allUserIds = rosterData.Players.Select(p => p.UserId)
+			.Concat(rosterData.Coaches.Select(c => c.UserId))
+			.Concat(rosterData.Staff.Select(s => s.UserId))
+			.Distinct()
+			.ToList();
+
+		// Validate all users exist and resolve to database IDs
+		var userIdMap = new Dictionary<UserIdentifier, long>();
+		foreach (var userId in allUserIds)
```

Comment:
This very inefficient, can we create a new extension called `WithIdentifiers` which will take an `IEnumerable<UserIdentifier>` and filter the users correctly based on those identifiers?